src/services/date-formatter.ts

```diff
@@ -0,0 +1,10 @@
+
+export abstract class DateFormatter {
```

Comment:
Why not just use date-fns?
https://date-fns.org/docs/Getting-Started

Reply:
This abstract class models the interface to be used by a `Format`, such that the format can be agnostic about what type of date formatting is used.
So *date-fns* could be used in a `DateFormatter` implementation.